aptos-move/aptos-vm/src/aptos_vm.rs

```diff
@@ -2159,41 +2125,18 @@ impl VMValidator for AptosVM {
         }
 
         if !self.features.is_enabled(FeatureFlag::WEBAUTHN_SIGNATURE) {
-            // exhaustive check TransactionAuthenticator in case variants are added later
-            match transaction.authenticator_ref() {
-                TransactionAuthenticator::Ed25519 { .. }
-                | TransactionAuthenticator::MultiEd25519 { .. } => {},
-                TransactionAuthenticator::MultiAgent {
-                    sender,
-                    secondary_signers,
-                    ..
-                } => {
-                    let is_webauthn = matches_webauthn_authenticator(sender)
-                        || includes_webauthn_authenticator(secondary_signers);
-                    if is_webauthn {
-                        return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING);
-                    }
-                },
-                TransactionAuthenticator::FeePayer {
-                    sender,
-                    secondary_signers,
-                    fee_payer_signer,
-                    ..
-                } => {
-                    let is_webauthn = matches_webauthn_authenticator(sender)
-                        || includes_webauthn_authenticator(secondary_signers)
-                        || matches_webauthn_authenticator(fee_payer_signer);
-                    if is_webauthn {
-                        return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING);
+            match transaction.authenticator_ref().to_single_key_authenticators() {
+                Ok(sk_authenticators) => {
+                    for authenticator in sk_authenticators {
+                        if let AnySignature::WebAuthn { .. } = authenticator.signature() {
+                            return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING)
+                        }
                     }
                 },
-                TransactionAuthenticator::SingleSender { sender } => {
-                    let is_webauthn = matches_webauthn_authenticator(sender);
-                    if is_webauthn {
-                        return VMValidatorResult::error(StatusCode::FEATURE_UNDER_GATING);
-                    }
+                _ => {
+                    return VMValidatorResult::error(StatusCode::INVALID_SIGNATURE);
```

Comment:
what's this? I think if it is this case, we get nothing...

Reply:
Well what should we return?  In practice this function would Err because of the breaking of invariants in MultiKey.  

If we cannot successfully fetch the authenticators for the signed transaction for me it implies that the signature is invalid (as is the case with MultiKey).

Is there another error we can return?

Reply:
I guess we could just do nothing, it'll get caught down the line.